### PR TITLE
List networks

### PR DESCRIPTION
**Pull request by ntavares**
Hey all, maybe you're interested in this, it's mostly a script to cycle the networks based on some criteria , although I had to touch the core a bit.

### CODE REVIEW

**Comment by remibergsma** [comment]
Thanks @ntavares will have a look at it!

Could you squash the commits into one and maybe add a few lines to the README file? :-)

**Comment by ntavares** [comment]
@remibergsma squashed. Also added README.md, sorry I missed it entirely. For the record, the squashed commit has 2 fixes and 1 new filter.

File: listNetworkVRs.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    global opFilterNoRR
+    opFilterNoRR = None
+
+    # Usage message
+    help = "Usage: ./" + os.path.basename(__file__) + ' [options] ' + \
+        '\n  --config-profile -c <profilename>\t\tSpecify the CloudMonkey profile name to get the credentials from (or specify in ./config file)' + \
+        '\n  --debug\t\t\t\t\tEnable debug mode' + \
+        '\n  --restart -r\t\t\t\t\tRestarts network w/ cleanup=True' + \
+        '\n    --type <networkType> \t\t\tApplies filter to operation, Possible networkTypes: Isolated,Shared,VPC,Isolated' + \
+        '\n    --onlyNoRR \t\t\t\t\Selects only non-redudant VR networks' + \
```
Comment (remibergsma) [inline listNetworkVRs.py]: One tab is missing a 't' here. Same on line 59 below.

screen shot 2016-03-13 at 18 18 58 pm

File: listNetworkVRs.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    
+    # Empty line
+    print
+    t = PrettyTable(["#", "Network", "Type", "ID", "Domain", "State", "Redundant?", "RestartReq?", "VRs"])
+    #t.align["VM"] = "l"
+
+    for n in networkData:
+        counter = counter + 1
+
+        t.add_row([counter, n['name'], n['type'], n['id'], n['domain'], n['state'], n['rr_type'], n['restartrequired'], n['vrs']])
```
Comment (remibergsma) [inline listNetworkVRs.py]: Would be nice if you could print the VRs a bit prettier ;-)

screen shot 2016-03-13 at 18 23 01 pm

**Comment by remibergsma** [comment]
@ntavares Thanks again, have been playing a bit with it and have a couple of questions.

The detection of redundant networks seems not to be OK. I happen to have zero redundant networks (it also displays a single router) and yet it thinks it's redundant:

python listNetworkVRs.py --onlyRR
screen shot 2016-03-13 at 18 29 15 pm

python listNetworkVRs.py --onlyNoRR
screen shot 2016-03-13 at 18 29 49 pm

Finally, a nice feature would be to select a single network to restart instead of a big collection. If you could add a switch like '-n' one can provide the name of the network to restart.

Thanks,
Remi

**Comment by ntavares** [comment]
Hi Remi,

Thanks for the feedback! I believe I found the problem, it was a missed assumption, apparently. Can you test it again?

I also changed the help page and the VRs printing.

As for the '-n'... funny as I've been thinking about the same... I will probably add support for it later on.
-NT

**Comment by ntavares** [comment]
Decided to implement -n anyway :-)
-NT

**Comment by ntavares** [comment]
Apparently there is a bug with what the API is reporting - a network might report capability RedundantRouter=true when its network offering doesn't have it. We're trying to track this down on CS. For now, commit ac8a7d7 will try to fetch that information from the network offering.

A lot more has been added. You guys might want to merge listDomains with listDomainsExt.

**Pull request closed without merging**